doc/releases/changelog-dev.md

```diff
@@ -6,6 +6,9 @@
 
 <h3>Improvements 🛠</h3>
 
+* Improve unit testing of `apply_operation` with `MidMeasureMP` and `Conditional`.
+  [(#6118)](https://github.com/PennyLaneAI/pennylane/pull/6118)
```

Comment:
We don't usually add changelog entries for changes to just testing.

Reply:
I can remove it, if we don't. 👍